state/state.go

```diff
@@ -1060,139 +1060,19 @@ func (st *State) AddApplication(args AddApplicationArgs) (_ *Application, err er
 	if args.Storage == nil {
 		args.Storage = make(map[string]StorageConstraints)
 	}
-	im, err := st.IAASModel()
```

Comment:
starting to get a bit hairy to keep everything paged in at once, can you please split these core State changes into its own PR?